modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceQuickAddToCart.testcase

```diff
@@ -0,0 +1,677 @@
+		}
+
+		task ("And results are shown") {
+			AssertElementPresent(
+				key_productSku = "MIN93015",
+				locator1 = "CommerceAccelerators#QUICK_MINI_CART_SEARCH_RESULT_BUTTON");
+		}
+
+		task ("And the buyer user can select the SKU") {
+			Click(
+				key_productSku = "MIN93015",
+				locator1 = "CommerceAccelerators#QUICK_MINI_CART_SEARCH_RESULT_BUTTON");
+		}
+
+		task ("Then the selected SKU can be added to the cart") {
+			Click(locator1 = "CommerceAccelerators#QUICK_MINI_CART_ADD_TO_CART_BUTTON");
+
+			AssertElementPresent(
+				key_cartItemSku = "MIN93015",
+				locator1 = "CommerceAccelerators#MINI_CART_ITEM_SKU");
+		}
+
+		task ("And an order is automatically created") {
+			CommerceNavigator.gotoMiniumSidebarItem(menuItem = "Pending Orders");
+
+			Button.click(button = "View");
+
+			CommerceOrders.viewOrdersDetails(
+				productListPrice = "50.00",
+				productName = "ABS Sensor",
+				productQuantity = "1",
+				productSku = "MIN93015",
+				productTotalPrice = "50.00");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-10389. Buyer user cannot select invalid SKUs on quick add to cart"
+	@priority = "4"
+	test CannotSelectInvalidSkusOnQuickAddToCart {
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+		property portal.acceptance = "false";
+
+		task ("Given an account with a buyer user is created") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And buyer user logs in") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("When the buyer opens the mini-cart") {
+			FrontStore.addInitializerProductToCart(
+				productName = "U-Joint",
+				productQuantity = "1");
+
+			// To be removed when COMMERCE-10472 is fixed
+
+			CommerceNavigator.gotoMiniCart();
+
+			Button.click(button = "Remove All Items");
+
+			// To be removed when COMMERCE-10472 is fixed
+
+			Button.click(button = "Yes");
+
+			// To be removed when COMMERCE-10472 is fixed
+
+		}
+
+		task ("And searches for an invalid sku on the quick add-to-cart") {
+			Type(
+				locator1 = "Publications#KEYWORD_TEXTFIELD",
+				value1 = "MIN99999");
+		}
+
+		task ("Then no results are shown") {
+			AssertElementNotPresent(
+				key_productSku = "MIN",
+				locator1 = "CommerceAccelerators#QUICK_MINI_CART_SEARCH_RESULT_BUTTON");
+		}
+
+		task ("And the search bar color is changed to red") {
+			KeyPress(
+				locator1 = "Publications#KEYWORD_TEXTFIELD",
+				value1 = "\ENTER");
```

Comment:
Does the keyboard keys work with the quick-add-to-cart? As far as I remember they don't.

Reply:
There's the bug COMMERCE-10474 when pressing the enter keyboard button on the searchbar but I'm using it in this test anyway because in this particular case the bug doesn't affect it. I'm gonna add a comment for this, thank you